src/UI/Components/Diagram/init.ts

```diff
@@ -181,66 +181,54 @@ export default function diagramInit(
 
   paper.on("link:connect", (linkView: dia.LinkView) => {
     //only id values are stored in the linkView
+
     const source = linkView.model.source();
     const target = linkView.model.target();
-    let didSourceChanged = false;
-    let didTargetChanged = false;
-
-    const sourceCell = graph.getCell(
-      source.id as dia.Cell.ID,
-    ) as ServiceEntityBlock;
-    const targetCell = graph.getCell(
-      target.id as dia.Cell.ID,
-    ) as ServiceEntityBlock;
-
-    if (
-      sourceCell.get("isEmbedded") &&
-      sourceCell.get("isEmbeddedTo") !== null
-    ) {
-      sourceCell.set("isEmbeddedTo", targetCell.id);
-      didSourceChanged = true;
-    }
-    if (
-      targetCell.get("isEmbedded") &&
-      targetCell.get("isEmbeddedTo") !== null
-    ) {
-      targetCell.set("isEmbeddedTo", sourceCell.id);
-      didTargetChanged = true;
-    }
-
-    const sourceRelations = sourceCell.getRelations();
-    const targetRelations = targetCell.getRelations();
-    const targetName = targetCell.getName();
-    const sourceName = sourceCell.getName();
-
-    if (sourceRelations) {
-      const doesSourceHaveRule = connectionRules[sourceName].find(
-        (rule) => rule.name === targetName,
-      );
 
-      if (doesSourceHaveRule) {
-        sourceCell.addRelation(targetCell.id as string, targetName);
-        didSourceChanged = true;
+    const checkUpdated = (sourceElement, targetElement) => {
```

Comment:
I see it's a bit better, but still confusing, why do we need to execute in both directions in lines 230 and 231? Can you also add some docs strings to the method?

Reply:
Because we can make the connections from the both sides, so we don't know which instance can get the data about being embedded/relatedTo, and previous version did that both way but there was some repetition, so avoiding repetition we need to check now from both perspectives, otherwise we have identical code but wrapped into extra function body.

I'll add docstring

Reply:
What do you mean, we don't know which instance gets the data, don't you have that information 🤔 You know the two elements that are being concerned by the event, you have the target, and the source, two instances the should get an update. Now, the way you are using  this, you are mixing up source and target... 

I'm lost on the reasoning to be fair, maybe the Kind field in the typing would already resolve some of the complexity of all those if cases throughout the file(s). 

Extracting logic doesn't mean you have to plainly copy some content into a separate method. I made the example previously, as a direction to try to explain what I meant, I didn't have all the pieces of the reasoning. Sometimes it helps to write out in bullet points what you are trying to achieve. 

For me, the two instances get an event, either it's adding something, or removing something. Then I see what kind of thing I need to add or remove, and update accordingly.

Reply:
This part of code checks which functions should get information about connection, as it's only stored in the children, I don't think that kind would help as theoretically both of them can be the same type and only one should get the data, so that's whats was done here.

We were checking both source and target, as depending from which instance we started it can be swapped on every attempt, that's why there was code repetition, we were checking the same thing on both instances, so as we now removed that into non-repetitive function we need to check it twice, for source and target

Reply:
But aren't you already doing the check on both directions in lines 213 and further? 
Honestly, I'm drowning in the amount of nested if cases all over, I can't keep track of what's happening. 
Right now, the only thing that got removed were those boolean values, but the repetition and complexity is still there. 

Lets discuss this tomorrow, have a call about this PR, it will be more efficient than discussing here.